app/bundles/DynamicContentBundle/Controller/DynamicContentController.php

```diff
@@ -479,12 +479,17 @@ public function deleteAction($objectId)
             $model  = $this->getModel('dynamicContent');
             $entity = $model->getEntity($objectId);
 
-            if (null === $entity) {
+            if (!$this->get('mautic.security')->isGranted('dynamiccontent:dynamiccontents:deleteown')
+                && !$this->get('mautic.security')->isGranted('dynamiccontent:dynamiccontents:deleteother')) {
```

Comment:
I'm wondering how the code can know whether the entity can be deleted based on "own" or "other" permissions when the security service doesn't know who created the entity. Shouldn't we use the `hasEntityAccess` method instead?

Isn't that exact check slightly bellow this one on line 493?

Reply:
Hi @escopecz, 
on lines 482, 483 the code checks if the logged in user has general permission to delete their own or other dynamic content.
If not, the user should not know if the entity exists or not. It should get 403, not 401.

In line 493 there is a checks base on existting entity Id.

Reply:
Is it necessary to check it twice? It seems like a code duplication to me.

Reply:
Ok, so how do we check if we can return 401 based on validation with the hasEntityAccess at line 493?

Reply:
Here is an example how "not found" is implemented for landing pages:

https://github.com/mautic/mautic/blob/5.x/app/bundles/PageBundle/Controller/PageController.php#L496-L507

Reply:
But this example does not check whether the user has permission to find out whether the entity exists or not.

Reply:
No entity in Mautic checks whether the user has permission to find out whether the entity exists or not. I wouldn't over-complicated this :)

Reply:
Yes, secuirity issues are sometimes overcomplicated and the Department of Security gives many absurd points. 
As you wish, I have removed this access checking :)